File: lab/python/POO/animal.py
from abc import ABC, abstractmethod


class Animal(ABC):
    @abstractmethod
    def falar(self):
        return


class Cachorro(Animal):
    def falar(self):
        return "Au au"


class Gato(Animal):
    def falar(self):
        return "Miau"


cachorro_carvao = Cachorro()
print("Cachorro:", cachorro_carvao.falar())

gato_butters = Gato()
print("Gato:", gato_butters.falar())

File: lab/javascript/POO/animal.js
// Classe base
class Animal {
  falar() {
    throw new Error("O método 'falar()' deve ser implementado pela subclasse.");
  }
}

// Subclasse Cachorro
class Cachorro extends Animal {
  falar() {
    return "Au au";
  }
}

// Subclasse Gato
class Gato extends Animal {
  falar() {
    return "Miau";
  }
}

// Teste
const cachorro = new Cachorro();
console.log("Cachorro:", cachorro.falar());

const gato = new Gato();
console.log("Gato:", gato.falar());
